discovery_engine_core/tooling/bin/newscatcher.rs

```diff
@@ -51,34 +52,33 @@ async fn main() -> Result<()> {
         .await
         .context("Failed to create download directory. Does it already exist?")?;
 
-    let client = Client::new(token, url);
+    let provider = NewscatcherHeadlinesProvider::new(url, token);
     let market = Market {
         lang_code: "en".to_string(),
         country_code: "US".to_string(),
     };
 
     // This is updated every iteration, based on the response from Newscatcher. So in reality,
     // we'll be fetching more than one page.
-    let mut total_pages = 1;
     let mut page = 1;
-    while page <= total_pages {
-        println!("Fetching page {} of {}", page, total_pages);
+    loop {
+        println!("Fetching page {}", page);
         let params = HeadlinesQuery {
-            common: CommonQueryParts {
-                market: Some(&market),
-                page_size: 100,
-                page,
-                rank_limit: RankLimit::LimitedByMarket,
-                excluded_sources: &[],
-            },
+            market: &market,
+            page_size: 100,
+            page,
+            rank_limit: RankLimit::LimitedByMarket,
+            excluded_sources: &[],
             trusted_sources: &[],
             topic: None,
             max_age_days: None,
         };
-        let raw_response = client.query_newscatcher(&params).await.unwrap();
-        total_pages = raw_response.total_pages;
+        let articles = provider.query_headlines(&params).await.unwrap();
+        if articles.is_empty() {
+            break;
+        }
 
-        let content = serde_json::to_string_pretty(&raw_response.articles)?;
+        let content = serde_json::to_string_pretty(&articles)?;
         tokio::fs::write(
             format!("./headlines_download/page_{:03}.json", page),
             content,
```

Comment:
:thinking: didn't we decide to remove it as it doesn't store what is received from the endpoint, but already fills in defaults etc. (and now also everything used for the to-generic conversion)?

Reply:
I still think it's useful for debugging, so I see no immediate need for removing it. iirc there were other people in the last discussion who felt similarly. If you feel strongly about removing this part of `tooling`, I think it would be nice if you brought this discussion to the daily and then recorded the outcome in the file as a comment :)